[ignition-cmake2] Add new port 🤖

traversaro opened the pull request:
Add a port for ignition-cmake2. As discussed in https://github.com/microsoft/vcpkg/pull/7781, different major version of ignition robotics libraries (https://ignitionrobotics.org/) can be installed side by side, so each new major version is added as a new port. 

In particular, ignition-cmake2 is a dependency for Gazebo 11 (http://gazebosim.org/blog/gazebo11) and for Ignition Robotics Citadel (that contains Ignition Gazebo 3, see https://www.openrobotics.org/blog/2019/12/11/ignition-citadel-released).

As far as I was able to test, this port should support all the triplets.

Related issue:
* https://github.com/microsoft/vcpkg/issues/8014#issuecomment-578431035

traversaro committed:
ignition-modularscripts: move CMake config files only if necessary

Some ignition libraries (such as ignition-cmake2) already install the
CMake config file in the location required by vcpkg

traversaro committed:
Add new port ignition-cmake2

NancyLi1013 commented on ports/ignition-cmake2/portfile.cmake:
```diff
@@ -0,0 +1,18 @@
+include(vcpkg_common_functions)
```
This function is no longer needed.
Could you please remove it?

traversaro replied:
Done.

NancyLi1013 commented on ports/ignition-modularscripts/CONTROL:
```diff
@@ -1,3 +1,3 @@
 Source: ignition-modularscripts
-Version: 2019-09-11
+Version: 2020-02-10
 Description: Vcpkg helpers to package ignition libraries
```
Could you please add homepage here?

traversaro replied:
This is a vcpkg-only port of scripts that help package ignition libraries, but it is not code that is provided by the ignition project at all, so I think adding an homepage would be misleading. 

Furthermore, similar ports for boost and  qt do not provide an homepage: 
* https://github.com/microsoft/vcpkg/blob/master/ports/qt5-modularscripts/CONTROL
* https://github.com/microsoft/vcpkg/blob/master/ports/boost-modular-build-helper/CONTROL

If a homepage is strictly required, we can add either the vcpkg one ( https://github.com/microsoft/vcpkg )  or the ignition libraries one ( https://ignitionrobotics.org/).

JackBoosY replied:
No need to add homepage in this case.

NancyLi1013 commented on ports/ignition-modularscripts/ignition_modular_library.cmake:
```diff
@@ -8,19 +8,21 @@ function(ignition_modular_build_library NAME MAJOR_VERSION SOURCE_PATH CMAKE_PAC
+    endif()
 
 
     # Post-build test for cmake libraries 
```
` vcpkg_test_cmake()` is deprecated. Could you please remove it?

traversaro replied:
Done.

NancyLi1013 commented on ports/ignition-cmake2/CONTROL:
```diff
@@ -0,0 +1,7 @@
+Version: 2.1.1
+Homepage: https://ignitionrobotics.org/libs/cmake
+Description: CMake helper functions for building robotic applications
+Build-Depends: ignition-modularscripts
```
The following space lines should be removed.

traversaro replied:
Done.

traversaro replied:
Done.

traversaro commented:
The CI now fails due to a qt5-tools regression, but I am not sure how this is related to the PR.

NancyLi1013 commented:
It seems that there is something wrong with Linux platform. 
The regression should be not related to this PR.
I have re run it. Please wait for some time.

NancyLi1013 commented:
/azp run

NancyLi1013 commented:
/azp run

NancyLi1013 commented:
/azp run

traversaro commented:
Hi @NancyLi1013 , I think I addressed all your comments, so I am not sure why the PR is listed under  "waiting for response"  . Let me know if I need to do something, thanks!

NancyLi1013 commented:
Hi @traversaro 

Sorry for the late response.
Since the CI test results were not all passed when I checked them before.

JackBoosY commented on ports/ignition-cmake2/portfile.cmake:
```diff
@@ -0,0 +1,16 @@
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib ${CURRENT_PACKAGES_DIR}/debug)
+
+# Install custom usage
+configure_file(${CMAKE_CURRENT_LIST_DIR}/usage ${CURRENT_PACKAGES_DIR}/share/${PORT}/usage @ONLY)
```
Miss copyright?

traversaro replied:
This is handled by `ignition_modular_library` CMake function in https://github.com/microsoft/vcpkg/blob/ca6e6d09ce1cbf956e94ca01079ad3f778377a8c/ports/ignition-modularscripts/ignition_modular_library.cmake#L33 .

strega-nil commented:
Alright, looks good to me!

traversaro commented:
Thanks a lot @strega-nil @NancyLi1013 @JackBoosY !
